e2e/test/scenarios/joins/reproductions/27380-dashboard-drops-joined-fields-on-zoom-in.cy.spec.js

```diff
@@ -1,52 +1,116 @@
+      },
+      dataset: true,
+    };
+    cy.createNativeQuestion(nativeQuestionDetails, {
+      wrapId: true,
+      visitQuestion: true,
+    });
+    openQuestionActions();
+
+    // Couldn't find a way to do this simply via API
+    popover().findByText("Edit metadata").click();
+
+    // We're on the first column `ID`
+    mapColumnTo({ table: "Orders", column: "ID" });
+
+    openColumnOptions("CREATED_AT");
+    mapColumnTo({ table: "Orders", column: "Created At" });
+
+    cy.button("Save changes").click();
+
+    cy.get("@questionId").then(questionId => {
+      const questionDetails = {
+        name: "27380 should not cause 29517",
+        query: {
+          "source-table": `card__${questionId}`,
+          aggregation: [["count"]],
+          breakout: [
+            [
+              "field",
+              "CREATED_AT",
+              {
+                "temporal-unit": "month",
+                "base-type": "type/DateTime",
+              },
+            ],
+          ],
+        },
+        display: "line",
+      };
+      cy.createQuestion(questionDetails, { visitQuestion: true });
+    });
+
+    // Doesn't really matter which 'circle" we click on the graph
+    cy.get("circle").last().realClick();
+
+    popover().findByText(`See these ${modelName}s`).click();
+
+    queryBuilderMain().findByText("Showing 344 rows");
+  });
 
+  it("should not drop fields from joined table on dashboard 'zoom-in' (metabase#27380)", () => {
+    const questionDetails = {
+      query: {
+        "source-table": ORDERS_ID,
+        aggregation: [["count"]],
+        breakout: [
+          [
+            "field",
+            PRODUCTS.CREATED_AT,
+            { "source-field": ORDERS.PRODUCT_ID, "temporal-unit": "month" },
+          ],
+        ],
+      },
+      display: "line",
+    };
     cy.createQuestionAndDashboard({ questionDetails }).then(
       ({ body: { dashboard_id } }) => {
         visitDashboard(dashboard_id);
       },
     );
-  });
 
-  it("should not drop fields from joined table on dashboard 'zoom-in' (metabase#27380)", () => {
     // Doesn't really matter which 'circle" we click on the graph
     cy.get("circle").last().realClick();
     // eslint-disable-next-line no-unscoped-text-selectors -- deprecated usage
-    cy.findByText("Zoom in").click();
+    cy.findByText("See this month by week").click();
     cy.wait("@dataset");
 
     // Graph should still exist
     // Let's check only the y-axis label
-    cy.get("y-axis-label").invoke("text").should("eq", "Count");
+    cy.get(".y-axis-label").invoke("text").should("eq", "Count");
 
     cy.icon("notebook").click();
     // eslint-disable-next-line no-unscoped-text-selectors -- deprecated usage
     cy.findByText("Pick a column to group by").should("not.exist");
     // eslint-disable-next-line no-unscoped-text-selectors -- deprecated usage
-    cy.findByText(/Products? → Created At: Month/);
+    cy.findByText("Product → Created At: Week");
```

Comment:
@WiNloSt was there a specific reason why you decided to remove the regex? IIRC, the pluralization can sometimes backfire so using regex is a way to guard against that.

Reply:
I believe generally we [want to use a simple plain string as much as possible](https://www.notion.so/metabase/43-Update-Style-Guide-2bcbc77ff4f34c9bb74a6623db22c9d6?pvs=4#9d35175600c549cc91f8bfcce5bdafd3). I don't know what you meant by
> the pluralization can sometimes backfire

As the same test should render the same pluralization right? Using regex gives us less confidence because we don't know if this is the whole string we want to match. But there are some situations when we want to use regex:
1. We really need to partially match strings e.g. when they also contain icons
2. Or for really long strings that we don't really care about every bit of detail.

Reply:
> As the same test should render the same pluralization right?

Not really.
Unless you set up the question details manually and hard coded an alias. I don't remember the details but it's somewhere in the back of my mind that this can sometimes render as Product and other times as Products.

Reply:
👆 Sounds like you mean each test run could result in a different pluralization? I think that's really bad. Can we leave this as is and I can change this back to regex if it flakes?